plugins/entity_points_converter.py

```diff
@@ -280,26 +284,28 @@ def csv_to_json_gen(file_, file_type) -> Iterator[dict]:
     # Yield all elements once in json format
     yield first_ent
     for ent in gen:
-        ent = ent._asdict()
         yield {"ID": ent["ID"], "href": ent["href"], "point": get_point(ent)}
 
 
 def json_to_json_gen(file_, file_type) -> Iterator[dict]:
     gen = load_entities(file_, mimetype=file_type)
-
     # Yield the first element twice. This will be used to later get the dimensionality of the point attribute
-    first_ent = next(gen)
-    yield first_ent
-
-    # Yield all elements once in json format
-    yield first_ent
-    for ent in gen:
-        yield ent
+    first = next(gen)
+    return itr_chain((first, first), gen)

```

Comment:
the chain function should be used at the location where it is required to inspect the first element. There is no reason why this iterator should accommodate for this need in a later iterator by yielding the first element twice! The current behaviour is surprising (evidence is the need for multiple long comments explaining this) and thus should be avoided (and it can be avoided).